tools/ci_build/github/azure-pipelines/linux-ci-pipeline.yml

```diff
@@ -144,7 +177,3 @@ jobs:
       searchFolder: '$(Build.BinariesDirectory)'
       testRunTitle: 'Unit Test Run'
     condition: succeededOrFailed()
-
-  - task: mspremier.PostBuildCleanup.PostBuildCleanup-task.PostBuildCleanup@3
```

Comment:
If we don't have this, we could soon run out of disk space if the machine is stateful. This is because the machine pool is used by multiple pipelines. For example,, if a machine was used to run this Linux-CPU-CI pipeline, but it didn't cleanup its build dir, then when it is used to run another pipeline it might not have enough free space.  Different pipelines use different build dirs.

Reply:
We have this. I just set it as the first step to make sure it would run as the last step.
Else it would run before Post-Job: ccache which would fail to upload cache data since the directory was clean. 
![image](https://user-images.githubusercontent.com/16190118/207649102-7bbe22d3-c0be-4dc1-9398-7970f0daba61.png)